### PR TITLE
Header & Sidebar prep for dark mode

### PR DESCRIPTION
**Pull request by kyleknighted**
## Description

Update Header and Sidebar colors to prepare for light/dark theme
Have pinned Sidebar meld into UI

## Motivation and Context

Theme All The Things

## Screenshots (if appropriate):
image

image

image
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/views/components/sidebar/NavItem.jsx [hunk 1]
```diff
@@ -13,80 +13,106 @@ type Props = {
...
   position: absolute;
-  padding: 5px 10px;
+  padding: 0 10px;
   left: 100%;
-  font-size: ${theme.fonts.size.body};
-  color: white;
-  background-color: ${theme.utils.contrastingColor(theme.colors.gray[10], 'AA')};
+  top: calc(50% - 13px);
+  height: 26px;
+  background-color: ${theme.colors.variant.lighter.info};
```
Comment (linuspahl) [inline graylog2-web-interface/src/views/components/sidebar/NavItem.jsx:32]: The blue looks a little too dark in my opinion.
image

While the next lighter blue would look a little too bright. What do you think about using the same styles like for an active pagination list item:

image

I used

```
  background-color: ${theme.colors.variant.lightest.info};
  border-top: 1px solid ${theme.colors.variant.lighter.info};
  border-right: 1px solid ${theme.colors.variant.lighter.info};
  border-bottom: 1px solid ${theme.colors.variant.lighter.info};
```

for the title + 
```
 background-color: ${isSelected ? theme.colors.variant.lighter.default : theme.colors.variant.lightest.info};
 border-bottom: 1px solid ${theme.colors.variant.lighter.info};
 border-left: 1px solid ${theme.colors.variant.lighter.info};
```
for Container -> IconWrapper -> ::after

File: graylog2-web-interface/src/views/components/sidebar/SidebarNavigation.jsx [hunk 2]
```diff
@@ -12,16 +12,18 @@ type Props = {
...
-  color: ${theme.colors.gray[80]};
-  box-shadow: 3px 0 3px rgba(0, 0, 0, 0.25);
+const Container: StyledComponent<{isOpen: boolean, sidebarIsPinned: boolean}, ThemeInterface, HTMLDivElement> = styled.div(({ isOpen, sidebarIsPinned, theme }) => css`
+  background: ${theme.colors.global.navigationBackground};
+  color: ${theme.utils.contrastingColor(theme.colors.global.navigationBackground, 'AA')};
+  box-shadow: ${(sidebarIsPinned && isOpen) ? 'none' : `3px 3px 3px ${theme.colors.global.navigationBoxShadow}`};
   width: 50px;
   height: 100%;
   position:relative;
+  z-index: 1031;
```
Comment (linuspahl) [inline graylog2-web-interface/src/views/components/sidebar/SidebarNavigation.jsx:26]: What do you think about adding the `z-index` to the theme config as well?

Comment (kyleknighted) [reply graylog2-web-interface/src/views/components/sidebar/SidebarNavigation.jsx:26]: Yes, there is a plan to do that but just have not gotten to it yet

File: graylog2-web-interface/src/views/components/sidebar/NavItem.jsx [hunk 3]
```diff
@@ -13,80 +13,106 @@ type Props = {
...
-      display: block;
-      border-right-color: currentColor;
-    }
+  ${IconWrap} {
+    overflow: ${isSelected ? 'unset' : 'hidden'};
 
-    :active {
-      &::after, > div {
-        display: none;
-      }
```
[the remark this answers is not in the document]
Comment (kyleknighted) [reply graylog2-web-interface/src/views/components/sidebar/NavItem.jsx:76]: After looking into this a bit, it is still here, it is just moved to different parts of the code

```
    ${Title} {
      display: ${(showTitleOnHover && !isSelected) ? 'flex' : 'none'};
    }

    ${IconWrap}::after {
      display: ${(showTitleOnHover) ? 'block' : 'none'};
    }
```

File: graylog2-web-interface/src/views/components/sidebar/NavItem.jsx (hunk 1 above)
Comment (kyleknighted) [reply graylog2-web-interface/src/views/components/sidebar/NavItem.jsx:32]: **regular**
image

**pinned**
image

File: graylog2-web-interface/src/views/components/sidebar/NavItem.jsx [hunk 4]
```diff
@@ -13,80 +13,106 @@ type Props = {
...
   position: relative;
+
+  ::after {
+    content: ' ';
+    position: absolute;
+    right: -9px;
+    top: calc(50% - 9px);
+    width: 18px;
+    height: 18px;
+    transform: rotate(45deg);
```
[the remark this answers is not in the document]
Comment (kyleknighted) [reply graylog2-web-interface/src/views/components/sidebar/NavItem.jsx:64]: Paired on this and updated

File: graylog2-web-interface/src/views/components/sidebar/NavItem.jsx (hunk 1 above)
Comment (linuspahl) [reply graylog2-web-interface/src/views/components/sidebar/NavItem.jsx]: I love it! Thanks for the effort.

File: graylog2-web-interface/src/components/graylog/Navbar.jsx [hunk 5]
```diff
@@ -4,282 +4,152 @@ import styled, { css } from 'styled-components';
...
+
+    &:hover,
+    &:focus {
+      color: ${theme.colors.variant.dark.default};
+      background-color: transparent;
+    }
+  }
+
+  .navbar-text {
+    color: ${theme.colors.global.textDefault};
```
[the remark this answers is not in the document]
Comment (linuspahl) [reply graylog2-web-interface/src/components/graylog/Navbar.jsx:23]: They look still different, but it is not a blocker for me, we can update them in a follow-up PR

**Pull request merged**